Classes/Projection/ProjectionManager.php

```diff
@@ -0,0 +1,214 @@
+use Neos\Cqrs\EventStore\EventStore;
+use Neos\Cqrs\EventStore\EventStreamFilter;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Cache\Frontend\VariableFrontend;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\Flow\Package\PackageManagerInterface;
+use TYPO3\Flow\Reflection\ClassReflection;
+use TYPO3\Flow\Reflection\ReflectionService;
+
+/**
+ * Central authority for managing projections
+ *
+ * @api
+ * @Flow\Scope("singleton")
+ */
+class ProjectionManager
+{
+    /**
+     * @Flow\Inject
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * @Flow\Inject
+     * @var ReflectionService
+     */
+    protected $reflectionService;
+
+    /**
+     * @Flow\Inject
+     * @var EventTypeResolver
+     */
+    protected $eventTypeResolver;
+
+    /**
+     * @Flow\Inject
+     * @var VariableFrontend
+     */
+    protected $projectionCache;
+
+    /**
+     * @Flow\Inject
+     * @var EventStore
+     */
+    protected $eventStore;
+
+    /**
+     * @var Projection[]
+     */
+    private $projections = [];
+
+    /**
+     * Register event listeners based on annotations
+     */
+    protected function initializeObject()
+    {
+        $this->projections = self::detectProjectors($this->objectManager);
+    }
+
+    /**
+     * @return Projection[]
+     */
+    public function getProjections()
+    {
+        return $this->projections;
+    }
+
+    /**
+     * @param string $projectionIdentifier
+     * @return void
+     */
+    public function replay(string $projectionIdentifier)
+    {
+        $eventTypes = [];
+        $fullProjectionIdentifier = $this->normalizeProjectionIdentifier($projectionIdentifier);
+        $projection = $this->projections[$fullProjectionIdentifier];
+        $projectorClassName = $projection->getProjectorClassName();
+        foreach (get_class_methods($projectorClassName) as $methodName) {
+            preg_match('/^when.*$/', $methodName, $matches);
+            if (!isset($matches[0])) {
+                continue;
+            }
+            $parameters = array_values($this->reflectionService->getMethodParameters($projectorClassName, $methodName));
+            $eventTypes[] = $this->eventTypeResolver->getEventTypeByClassName($parameters[0]['class']);
+        }
+        $filter = new EventStreamFilter();
+        $filter->eventTypes = $eventTypes;
+
+        $projector = $this->objectManager->get($projectorClassName);
+        /** @var EventTransport $eventTransport */
+        foreach ($this->eventStore->get($filter) as $eventTransport) {
+            $listenerMethodName = 'when' . $this->eventTypeResolver->getEventShortType($eventTransport->getEvent());
+            call_user_func([$projector, $listenerMethodName], $eventTransport->getEvent(), $eventTransport->getMetadata());
+        }
+    }
+
+    /**
+     * @param string $projectionIdentifier
+     * @return void
+     */
+    public function catchup(string $projectionIdentifier)
```

Comment:
Public method for increasing projection revision? Not sure I get it... it's also not (yet) used anywhere